src/yb/util/atomic.h

```diff
@@ -444,6 +445,43 @@ class AtomicTryMutex {
   std::atomic<bool> locked_{false};
 };
 
+class AtomicTryMutexWithTimestamp {
+ public:
+  void unlock() {
+    auto unlock_timestamp = std::chrono::steady_clock::now();
+    auto duration =
+        std::chrono::duration_cast<std::chrono::milliseconds>(unlock_timestamp - lock_timestamp_)
+            .count();
+
+    if (duration > warning_threshold_ms_) {
+      LOG(WARNING) << "AtomicTryMutex locked for too long: " << duration << " ms";
+    }
+    mutex_.unlock();
+  }
+
+  bool try_lock() {
+    auto success_locked = mutex_.try_lock();
+    if (success_locked) {
+      lock_timestamp_ = std::chrono::steady_clock::now();
+    }
+    return success_locked;
+  }
+
+  bool is_locked() const {
+    return mutex_.is_locked();
+  }
+
+  // Set the threshold duration for warnings
+  void set_warning_threshold(long long threshold_ms) {
+    warning_threshold_ms_ = threshold_ms;
+  }
+
+ private:
+  AtomicTryMutex mutex_;
+  std::chrono::steady_clock::time_point lock_timestamp_;
+  long long warning_threshold_ms_ = 1000;  // Default threshold
```

Comment:
Please define the threashold as a GFlag,
long long warning_threshold_ms_ = 1000;
see
DEFINE_RUNTIME_int64(cdc_intent_retention_ms, 4 * 3600 * 1000,
    "Interval up to which CDC consumer's checkpoint is considered for retaining intents."
    "If we haven't received an updated checkpoint from CDC consumer within the interval "
    "specified by cdc_checkpoint_opid_interval, then CDC does not consider that "
    "consumer while determining which op IDs to delete from the intent.");

Reply:
For this, should we add the GFlag to atomic.h itself or move both the GFlag and the class definition to atomic.cc ?

Reply:
@ajd12342 Oh, the flag should be moved to atomic.cc